src/java.base/share/classes/java/time/format/DateTimeTextProvider.java

```diff
@@ -309,13 +309,7 @@ public Iterator<Entry<String, Long>> getTextIterator(Chronology chrono, Temporal
 
     private Object findStore(TemporalField field, Locale locale) {
         Entry<TemporalField, Locale> key = createEntry(field, locale);
-        Object store = CACHE.get(key);
-        if (store == null) {
-            store = createStore(field, locale);
-            CACHE.putIfAbsent(key, store);
-            store = CACHE.get(key);
-        }
-        return store;
+        return CACHE.computeIfAbsent(key, e -> createStore(e.getKey(), e.getValue()));
```

Comment:
If `createStore` can be static, the call site will only have to construct a constant lambda than having to pass `this` to construct a new instance on every call

Reply:
Well, if you _really_ want to noodle this for performance, you can also store a `this`-bound lambda in a `private final`  instance field, so then it's only created once too. I wouldn't put it past `javac` to do this, but I'd have to disassemble the bytecode of a little test program to see whether it's the case.

Reply:
@RogerRiggs is you comment about `DateTimeTextProvider.findStore` or  `WeekFields.of`?
There is no record creation here, in `DateTimeTextProvider.findStore`.

Reply:
@RogerRiggs in my view, it's not about performance per se… I'll sometimes help along these kinds of small fix PRs (that are more likely than not driven by IntelliJ code refactoring suggestions) if I think they lead to more idiomatic code in the JDK with no drawbacks. I think there's value in periodically revisiting existing code and improve it to use new language and library construct that have become available since – people study JDK source code, and I think it's important they see as good examples in there as possible. I do like seeing an old example of hand-rolled compute-if-absent being replaced by an actual `computeIfAbsent` call. Similarly, I think using records as composite keys is in general a better practice over simulating a composite key by creating a string representation of it. 

Yeah, it'll load additional code at runtime for that record's class, and yes, I'm aware records' `hashCode`/`equals`/`toString` are implemented by delegating to factory methods that heavily use method handle combinators. If records are meant as lightweight value-semantics composites, it should be okay to use them like this and if there are performance concerns with their use, then those concerns should be addressed by improving their performance. OTOH, MH combinators installed into constant call sites are well optimized by HotSpot these days… anyhow, a discussion for another day.

In this specific instance, the value domain of the keys is limited, but the number of method calls to `WeekFields.of` isn't. It could be benchmarked but if there's concerns, it's also okay to de-scope the record-as-key from this PR. I don't have strong feelings either way.

Reply:
I agree with @RogerRiggs that performance of JDK is very important. It was one of main motivation for removing `.get` call.
I'm not an expert in microbenchmarking, and it would require significant time/resources for me to investigate which is better.
So I decided to revert back to original proposal (without lambda and `computeIfAbsent`) in `WeekFields`. If you think that it's better to improve code, I think we can always fill a separate issue for it.

Reply:
Fair.